src/exodus/wallet.cpp

```diff
@@ -0,0 +1,90 @@
+#include "wallet.h"
+#include "wallet/wallet.h"
+#include "wallet/walletdb.h"
+#include "walletdb.h"
+
+namespace exodus {
+
+SigmaPublicKey Wallet::CreateSigmaMint(
+    uint32_t propertyId,
+    uint8_t denomination)
+{
+    SigmaPrivateKey key;
+
+    do {
+        key.Generate();
+    } while (!key.IsValid());
```

Comment:
We should make `SigmaPrivateKey::Generate()` to never generate invalid key instead.